presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java

```diff
@@ -62,12 +67,17 @@ public void testLength()
         assertFunction("LENGTH('')", 0);
         assertFunction("LENGTH('hello')", 5);
         assertFunction("LENGTH('Quadratically')", 13);
+        //
+        // Test length for non-ASCII
+        assertFunction("LENGTH('hello na\u00EFve world')", 17);
+        assertFunction("LENGTH('\uD801\uDC2Dend')", 4);
+        assertFunction("LENGTH('\u4FE1\u5FF5,\u7231,\u5E0C\u671B')", 7);
```

Comment:
Take a look at the constructor of https://github.com/facebook/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/TestingRowConstructor.java . This lets you register custom functions for the current test only. This way, you can produce illegal unicode sequences in tests.